src/myst/syntax/parser.cr

```diff
@@ -1,11 +1,19 @@
 require "./ast.cr"
+require "tempfile"
 
 module Myst
   class Parser < Lexer
     def self.for_file(source_file)
       new(File.open(source_file), source_file)
     end
 
+  def self.for_content(content)
```

Comment:
Instead of writing a temp file I think you can just write the content to an IO object, something along the lines of `Parser.new(IO::Memory.new(content), "")`.  That would avoid the need for this method.  Just my opinion!

Reply:
Good idea! Haven't worked to much with IOs other than std's and files

Reply:
@bmulvihill After looking into it some more, a search reveals :
``` 
#In src/myst/syntax/lexer.cr
line 43:      @source_file = File.expand_path(source_file)
line 68:      @current_token = Token.new(location: Location.new(@source_file, @row, @col))
```
The token#location seems to be very problematic and require a lot of refactoring.
I want to hear @faultyserver's thoughts on this

Reply:
@Jens0512 it should work, Ive parsed and interpreted code from a Memory IO before, how are you trying to do it?

Reply:
So far i've only looked through the uses of `source_file`, haven't actually tried to the `IO::Memory.new content` yet. I'll do so now, just found it a bit weird to leave the location which is passed around so much empty.

Reply:
The string is just the filename, so everything should still work, it will use "anonymous" I believe if no filename is provided

Reply:
`IO::Memory` worked just fine! (Just as you said) Guess the Location is not all that important, sorry for fuzzing over something unnecessary.